### PR TITLE
Add schema to support tangent plane WCS corrections

### PR DESCRIPTION
**Pull request by mcara**
CC: @nden Please review this PR.

### CODE REVIEW

File: jwst/transforms/schemas/tpcorr-0.7.0.yaml [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+          Roll angle.
+        type: number
+      matrix:
+        description: |
+          Generalized rotation matrix.
+        type: array
+        items:
+          type: array
+          items:
+            type: number
```
Comment (nden) [inline jwst/transforms/schemas/tpcorr-0.7.0.yaml]: This limits the schema to a list of lists. If you know that you are never going to pass a numpy array then it's fine. Or yo ucan pass a ndarray but make sure that you convert it to a list of lists. 

Alternatively you can us `anyOf` and add ndarray as an alternative.

In any case add a test to tags/tests which writes the model to a file and then reads it back in. There are examples in the test file.

Comment (mcara) [reply jwst/transforms/schemas/tpcorr-0.7.0.yaml]: @nden Thank you! I could not find any examples of how to do this but I changed the code `TPCorrType` to perform the conversion after/before serialization.

**Comment by mcara** [comment]
@nden I think I got the idea of `anyOf`. Many thanks!

**Comment by mcara** [comment]
@jdavies-st I think now you can try running the entire pipeline - hopefully serialization will work as expected. Thanks!

**Pull request merged**